Sources/Amplitude/Utilities/Diagonostics.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  Diagonostics.swift
+//  Amplitude-Swift
+//
+//  Created by Qingzhuo Zhen on 3/4/24.
+//
+
+import Foundation
+
+public class Diagnostics {
+    private var malformedEvents: [String]?
+    private var errorLogs: [String]?
+
+    init(){}
+
+    func addMalformedEvent(_ event: String) {
+        if malformedEvents == nil {
+            malformedEvents = [String]()
+        }
+        malformedEvents?.append(event)
+    }
+
+    func addErrorLog(_ log: String) {
+        if errorLogs == nil {
+            errorLogs = [String]()
+        }
+        errorLogs?.append(log)
+    }
+
+    func hasDiagnostics() -> Bool {
+        return (malformedEvents != nil && malformedEvents!.count > 0) || (errorLogs != nil && errorLogs!.count > 0)
+    }
+
+    /**
+     * Extracts the diagnostics as a JSON string.
+     * @return JSON string of diagnostics or empty if no diagnostics are present.
+     */
+     func extractDiagonostics() -> String {
```

Comment:
nit: rename to `toString()` or `toJson` to match convention?

Reply:
I use the extract on purpose as it will also clear out the current data in this function. I could rename it to extractDiagonosticsToStirng if that sounds better.

Reply:
Ah that makes sense. Thanks